casper/src/test/scala/coop/rchain/casper/genesis/DeployPaymentCostTest.scala

```diff
@@ -0,0 +1,257 @@
+package coop.rchain.casper.genesis
+
+import cats.implicits._
+import cats.{Apply, Id}
+import com.google.protobuf.ByteString
+import coop.rchain.casper.genesis.DeployPaymentCostTest._
+import coop.rchain.casper.genesis.contracts.{Faucet, PreWallet}
+import coop.rchain.casper.helper.HashSetCasperTestNode
+import coop.rchain.casper.protocol.{BlockMessage, Deploy, DeployData, PhloPrice}
+import coop.rchain.casper.util.ProtoUtil
+import coop.rchain.casper.util.rholang.RuntimeManager
+import coop.rchain.casper.{Created, HashSetCasperTest, MultiParentCasperImpl, Valid}
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.hash.{Blake2b256, Keccak256}
+import coop.rchain.crypto.signatures.{Ed25519, Secp256k1}
+import coop.rchain.models.Expr.ExprInstance._
+import coop.rchain.models.Var.VarInstance.{BoundVar, FreeVar, Wildcard}
+import coop.rchain.models.Var.WildcardMsg
+import coop.rchain.models._
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.rholang.interpreter.{accounting, Interpreter}
+import coop.rchain.rspace.Serialize
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.FlatSpec
+
+import scala.collection.immutable.BitSet
+
+class DeployPaymentCostTest extends FlatSpec {
+
+  "DeployPaymentCost" should "estimate cost of deploying payment code" in {
+    val node = HashSetCasperTestNode.standalone(genesis, validatorKeys.head)
+    import node.casperEff
+
+    val secKey = Base16.decode(secKeyString)
+    val pubKey = Base16.decode(pubKeyString)
+
+    // Create new wallet
+    val walletRetCh = "walletRet"
+    val newWalletName = deployAndCapture(
+      createWalletPar(pubKeyString, walletRetCh),
+      GString(walletRetCh),
+      node.runtimeManager
+    )
+    val walletAddress =
+      GPrivateBuilder(
+        ByteString.copyFrom(newWalletName.exprs.head.getEListBody.ps.head.ids.head.id.toByteArray)
+      )
+
+    // Register wallet in the Registry
+    val registerWalletTuple: Par = ETuple(Seq(GInt(1), walletAddress))
+    val registrySig              = Ed25519.sign(registerWalletTuple.toByteArray, secKey)
+    Ed25519.verify(registerWalletTuple.toByteArray, registrySig, pubKey)
+
+    val walletUriRet: Par = GPrivateBuilder()
+
+    val registerWalletPar = registerSigned(
+      registerWalletTuple,
+      ByteString.copyFrom(pubKey),
+      ByteString.copyFrom(registrySig),
+      walletUriRet
+    )
+
+    val walletUri = deployAndCapture(registerWalletPar, walletUriRet, node.runtimeManager)
+
+    val walletUriString = walletUri.exprs.head.getGUri
+
+    // Prepare wallet transfer
+    val nonce  = 0
+    val amount = 10
+    //TODO: use unforgeable names
+
+    val transferChannel  = "myWithdrawal"
+    val statusChannel    = "withdrawalStatus"
+    val transfer         = parse(s"""[$nonce, $amount, "$transferChannel"]""")
+    val transferBytes    = bytes(transfer)
+    val sigTransferBytes = Ed25519.sign(Blake2b256.hash(transferBytes), secKey)
```

Comment:
I think the signing can be done in rholang too?

Reply:
No, you can't. To sign you need private key and you should never put private key on the blockchain.